### PR TITLE
Orbe & Macatula: New game and Continue or Restart Function

### PR DESCRIPTION
**Pull request by johnorbz**
Contributors: John Orbe and Rosie Macatula
from line 240 to 274
Added "continue_or_restart" as suggested un the group chat.
Added "new_game" function

### CODE REVIEW

File: word-search.py [hunk 1]
```diff
@@ -237,4 +237,40 @@ def word_search():
...
+        else:
+            print("Invalid input. Please enter 'y' to continue or 'n' to restart.")
+
+def new_game():
+    while True: 
+        gridsize, timeroption = menu() #get grid size and timer option
+        word_search()  # start game
+
+
+        if timeroption == 4: #if untimed game is selected, ask the player if they want to continue or restart in between guesses
```
Comment (reggyball) [inline word-search.py:251]: based on word search process flow #47, user is be presented with game options 1. enter a word and 2. new game. code for entering word is missing and presenting options is missing po ata :(

File: word-search.py [hunk 2]
```diff
@@ -237,4 +237,40 @@ def word_search():
...
+            return True  #continue playing
+        elif option == 'n':
+            return False  #quit
+        else:
+            print("Invalid input. Please enter 'y' to continue or 'n' to restart.")
+
+def new_game():
+    while True: 
+        gridsize, timeroption = menu() #get grid size and timer option
+        word_search()  # start game
```
Comment (reggyball) [inline word-search.py]: Menu is also called inside the word search, hence when played user is prompted twice for the grid size and timer. Maybe the flow can be unified :))

Comment (johnorbz) [reply word-search.py]: yes, para hindi mag D.R.Y., actually na notice ko yan ginawa ko lang uli para di ma galaw yung existing codes.

Comment (johnorbz) [reply word-search.py]: yung word_search() i-omit na ilagay na sa new_game()

File: word-search.py (hunk 1 above)
Comment (johnorbz) [reply word-search.py:251]: I think nasa line 252 na under menu(), well I could be wrong.

File: word-search.py (hunk 2 above)
Comment (reggyball) [reply word-search.py]: I think it is okay to update the existing code specifically the word_search  para po hindi doble yung menu?  or like u said po omit word search and include it sa new_game :)

Comment (johnorbz) [reply word-search.py]: yes, let's also wait for vickie's call as well, since kaya nga may branches.

File: word-search.py (hunk 1 above)
Comment (reggyball) [reply word-search.py:251]: grid and timer option po yung nasa 252. Ito po yata yung game menu pag unlimited time pero di ko din po sure if scope ng function niyo

Screenshot 2024-11-30 215836

Comment (johnorbz) [reply word-search.py:251]: as per Vickie in group chat under ng task namin.

Comment (johnorbz) [reply word-search.py:251]: medyo iba lang yung nagawa ko dyan check line 240 to 248 pero same essence >> option to continue or not after every guess pag untimed game

File: word-search.py (hunk 2 above)
Comment (vdepasupillagazo) [reply word-search.py]: Thanks for reviewing @reggyball! @johnorbz I do agree that we can absorb word_search into the new_game function at this point. word_search was only meant to help us visualize the  flow and progress so far, anyway. Let's use new_game moving forward.

Comment (johnorbz) [reply word-search.py]: noted on this, will get it done within the day, thank you both.

**Pull request closed without merging**